pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -480,6 +489,12 @@ def reconciliate_processes(self, trace: str):
         if self.world_size < 2:
             return
 
+        # If the cluster environment creates the process, allow the scheduler / parent process
+        # to perform the process termination
+        cluster_env = self.cluster_environment
+        if cluster_env is not None and (cluster_env.creates_children() and not self._has_called_call_children_scripts):
+            return
+
```

Comment:
do we need to check the cluster_env here at all? ideally doesn't this always hold `cluster_env.creates_children() = not self._has_called_call_children_scripts`? this way it's clear: only if within the plugin lightning launched the processes, it proceeds with reconciliation. otherwise it doesn't.

```suggestion
        # If the cluster environment creates the process, allow the scheduler / parent process
        # to perform the process termination
        if not self._has_called_call_children_scripts:
            return
```